src/auth.js

```diff
@@ -0,0 +1,24 @@
+const { OAuth2Client } = require('google-auth-library');
+
+const CLIENT_ID =
+  '270040816063-djog7f8mpvt4m162ak4n04bjmftg1bhc.apps.googleusercontent.com';
+const client = new OAuth2Client(CLIENT_ID);
+
+const requireAuth = async (ctx, next) => {
+  const [_, idToken] = ctx.header.authorization.split(' ');
+  const ticket = await client.verifyIdToken({
+    idToken,
+    audience: CLIENT_ID,
+  });
```

Comment:
When I first tested this I had (what I think was) an old Google token.  It caused the `client.verifiyIdToken` call to return a `Token used too late` error which manifested as a `500` on the client-side.  Could be worth wrapping this in a `try/catch` to return a `401` if it fails instead.

Reply:
💯